python/llm/example/transformers/transformers_int4/whisper/readme.md

```diff
@@ -57,4 +58,61 @@ numactl -C 0-47 -m 0 python ./recognize.py
 Inference time: xxxx s
 -------------------- Output --------------------
 [" Mr. Quilter is the Apostle of the Middle classes and we're glad to welcome his Gospel."]
-```
\ No newline at end of file
+```
+
+
+## Example 2: Recognize Long Segment using `generate()` API
+In the example [long-segment-recognize.py](./long-segment-recognize.py), we show a basic use case for a Whisper model to conduct transcription using `pipeline()` API, with BigDL-LLM INT4 optimizations.
+### 1. Install
+We suggest using conda to manage environment:
+```bash
+conda create -n llm python=3.9
+conda activate llm
+
+pip install bigdl-llm[all] # install bigdl-llm with 'all' option
+pip install datasets soundfile librosa # required by audio processing
+```
+
+### 2. Run
+The Whisper model is intrinsically designed to work on audio samples of up to 30s in duration. For audio recordings longer than 30 seconds, it is possible to enable batched inference with `pipeline` method:
+```
+python ./long-segment-recognize.py --repo-id-or-model-path REPO_ID_OR_MODEL_PATH --audio-file PATH_TO_THE_AUDIO_FILE --language LANGUAGE --chunk-length CHUNK_LENGTH
+```
+
+Arguments info:
+- `--repo-id-or-model-path REPO_ID_OR_MODEL_PATH`: argument defining the huggingface repo id for the Whisper model to be downloaded, or the path to the huggingface checkpoint folder. It is default to be `'openai/whisper-medium'`.
+- `--audio-file PATH_TO_THE_AUDIO_FILE`: argument defining the path of the audio file to be recognized.
+- `--language LANGUAGE`: argument defining language to be transcribed. It is default to be `english`.
+- `--chunk-length CHUNK_LENGTH`: argument defining The maximum number of chuncks of sampling_rate samples used to trim and pad longer or shorter audio sequences. It is default to be 30.
```

Comment:
You forgot to add the updated argument info for chunk-length as well as batch-size here :)

Reply:
And it would be better to mention that the chunk-length should not be larger than 30s for whisper model (if I understand correctly)